extension/src-webview/views-rendering.tsx

```diff
@@ -170,3 +175,93 @@ export function renderHexagon(node: SNode): VNode {
     />;
 }
 
+/**
+ * Creates an And-Gate for {@code node}.
+ * @param node The node that should be represented by an And-Gate.
+ * @returns An And-Gate for {@code node}.
+ */
+export function renderAndGate(node: SNode): VNode {
+    const leftX = 0;
+    const midX = Math.max(node.size.width, 0) / 2.0;
+    const rightX = Math.max(node.size.width, 0);
+    const botY = Math.max(node.size.height, 0);
+    const midY = Math.max(node.size.height, 0) / 2.0;
+    const topY = 0;
+
+    const d = `M ${leftX}, ${midY} V ${botY} H ${rightX} V ${midY} C ${rightX}, ${midY} ${rightX}, ${topY} ${midX}, ${topY} ${leftX}, ${topY} ${leftX}, ${midY} ${leftX}, ${midY} Z`;
+    // 'M' + leftX + " " + midY + " V " + botY + " H " + rightX + " V " + midY + " C " + rightX + " " + midY + " " + rightX + " "
+    // + topY + " " + midX + " " + topY + " " + leftX + " " + topY + " " + leftX + " " + midY + " " + leftX + " " + midY + 'Z';
+
+    return <path
+        d={d}
+    />;
+}
+
+/**
+ * Creates an Or-Gate for {@code node}.
+ * @param node The node that should be represented by an Or-Gate.
+ * @returns An Or-Gate for {@code node}.
+ */
+export function renderOrGate(node: SNode): VNode {
+    const leftX = 0;
+    const rightX = Math.max(node.size.width, 0);
+    const midX = rightX / 2.0;
+    const botY = Math.max(node.size.height, 0);
+    const nearBotY = botY - 5;
+    const midY = Math.max(node.size.height, 0) / 2;
+    const topY = 0;
+    const d = `M${leftX},${midY} V ${botY}` + `C ${leftX}, ${botY} ${leftX+10}, ${nearBotY} ${midX}, ${nearBotY} ${rightX-10}, ${nearBotY} ${rightX}, ${botY} ${rightX}, ${botY}`
+    + `V ${midY} A ${node.size.width},${node.size.height-10},${0},${0},${0},${midX},${topY} A ${node.size.width},${node.size.height-10},${0},${0},${0},${leftX},${midY} Z`;
+
+    return <path
+        d={d}
+    />;
+}
+
+/**
+ * Creates an Kn-Gate for {@code node}.
+ * @param node The node that should be represented by an Kn-Gate.
+ * @returns An Kn-Gate for {@code node}.
+ */
+export function renderKnGate(node: SNode, k: number, n: number): VNode {
+    const leftX = 0;
+    const rightX = Math.max(node.size.width, 0);
+    const midX = rightX / 2.0;
+    const botY = Math.max(node.size.height, 0);
+    const nearBotY = Math.max(node.size.height, 0) - (Math.max(node.size.height, 0) / 10.0);
+    const midY = Math.max(node.size.height, 0) / 2;
+    const topY = 0;
+    const d = `M${leftX},${midY} V ${botY}` + `C ${leftX}, ${botY} ${leftX+10}, ${nearBotY} ${midX}, ${nearBotY} ${rightX-10}, ${nearBotY} ${rightX}, ${botY} ${rightX}, ${botY}`
```

Comment:
see above, all these constructions are quite obscure, maybe a comment explaining what shape is being made would be helpful